src/main/java/de/azapps/kafkabackup/sink/BackupSinkTask.java

```diff
@@ -101,10 +168,23 @@ public void open(Collection<TopicPartition> partitions) {
                     // The offset was not found, so rather than forcibly set the offset to 0 we let the
                     // consumer decide where to start based upon standard consumer offsets (if available)
                     // or the consumer's `auto.offset.reset` configuration
+
+                    // if we are in snapshot mode, then just start at zero.
```

Comment:
I don't think that we want it that way – this would cause a rebuild of the whole backup in case Kafka Connect crashes. I see where this comes from I am also not absolutely happy with the current behaviour. I think we should do that in an extra PR. Lets discuss it in #100 further.

Reply:
I'm not too happy about this either, but to me this seems reasonable in the case of snapshots. If kafka connect crashes, upon restart, the last offset will be retrieved from the file system, so the backup will resume. My line of thought for the zero offset is a 'new target folder' situation. The start offset from file system would be -1, but the consumer offset will not be 0. This is undesirable for a snapshot. There are other ways to fix this though.

Reply:
Ah sorry, I was confused by the diff view. missed the else branch. You are correct. With #100 I would even set the `context.offset(topicPartition, 0)` as the default behaviour even without the snapshotMode.